apps/server/src/modules/idp-console/idp-console.module.ts

```diff
@@ -0,0 +1,81 @@
+import { Module } from '@nestjs/common';
+import { ConsoleModule } from 'nestjs-console';
+import { CqrsModule } from '@nestjs/cqrs';
+import { ConsoleWriterModule } from '@infra/console';
+import { ConfigModule } from '@nestjs/config';
+import { HttpModule, HttpService } from '@nestjs/axios';
+import { Configuration } from '@hpi-schul-cloud/commons/lib';
+import { Logger, LoggerModule } from '@src/core/logger';
+import { MikroOrmModule } from '@mikro-orm/nestjs';
+import { DB_PASSWORD, DB_URL, DB_USERNAME } from '@src/config';
+import { SchulconnexRestClient } from '@infra/schulconnex-client';
+import { AccountModule } from '@modules/account';
+import { OauthAdapterService } from '@modules/oauth';
+import { defaultMikroOrmOptions } from '@modules/server';
+import { ALL_ENTITIES } from '@shared/domain/entity';
+import { UserService } from '@modules/user';
+import { UserRepo } from '@shared/repo';
+import { UserDORepo } from '@shared/repo/user/user-do.repo';
+import { RoleModule } from '@modules/role';
+import { IdentityManagementModule } from '@infra/identity-management';
+import { RegistrationPinModule } from '@modules/registration-pin';
+import {
+	SynchronizationEntity,
+	SynchronizationRepo,
+	SynchronizationService,
+	SynchronizationUc,
+} from '@modules/synchronization';
+import { IdpSyncConsole } from './idp-sync-console';
+
+@Module({
+	imports: [
+		ConfigModule.forRoot({ isGlobal: true }),
+		ConsoleModule,
+		ConsoleWriterModule,
+		LoggerModule,
+		CqrsModule,
+		MikroOrmModule.forRoot({
+			...defaultMikroOrmOptions,
+			type: 'mongo',
+			clientUrl: DB_URL,
+			password: DB_PASSWORD,
+			user: DB_USERNAME,
+			allowGlobalContext: true,
+			entities: [...ALL_ENTITIES, SynchronizationEntity],
+			debug: true,
+		}),
+		IdentityManagementModule,
+		HttpModule,
+		AccountModule,
+		RegistrationPinModule,
+		RoleModule,
+	],
+	providers: [
+		SynchronizationRepo,
+		SynchronizationService,
+		SynchronizationUc,
+		OauthAdapterService,
+		IdpSyncConsole,
+		UserDORepo,
```

Comment:
why do you need the Repos? cant you use the UserService?
And if you would need them, why would you need both of them??

Reply:
`UserService` doesn't want to work without the `UserDORepo`:
```
Error: Nest can't resolve dependencies of the UserService (UserRepo, ?, ConfigService, RoleService, AccountService, RegistrationPinService, Logger, EventBus). Please make sure that the argument UserDORepo at index [1] is available in the IdpConsoleModule context.
```
and without `UserRepo`:
```
Error: Nest can't resolve dependencies of the UserService (?, UserDORepo, ConfigService, RoleService, AccountService, RegistrationPinService, Logger, EventBus). Please make sure that the argument UserRepo at index [0] is available in the IdpConsoleModule context.
```
and as explained in https://github.com/hpi-schul-cloud/schulcloud-server/pull/4859#discussion_r1555991463 importing whole `UserModule` doesn't make the situation easier.

Reply:
You are right, if you add one thing as provider, you need to add everything. And if something changes, you are screwed. We made this mistake in the past, it never works out. A clean Dependency on the Module is the only way that works, and that means your application has to be able to use the Module as defined.

Either make it work, or change the UserModule to be able to be started in the way you use it, or split the UserModule so the core part can be used without additional dependencies.